python/ray/serve/_private/http_proxy.py

```diff
@@ -164,6 +165,12 @@ async def _send_request_to_handle(handle, scope, receive, send) -> str:
         return "500"
 
     if isinstance(result, (starlette.responses.Response, RawASGIResponse)):
+        if isinstance(result, starlette.responses.Response):
```

Comment:
hmm this makes me realize that my PR will complicate this a bit...

Reply:
I will let your pr checked in and then revisit here.